Automattic/wp-calypso
# Add/jetpack connect store

This PR adds to the redux store the reducers & associated actions related with jetpack-connect, so we can use the API new endpoint to to check an url and pass its status to the jetpack-connect main component.
With this changeset, jetpack-connect main component already checks the user-entered url and displays warnings based in its status. If the url has a non-connected jetpack, starts the connection process, and if it has no jetpack installed, shows the dialog explaining that the plugin need to be installed first.

Some screen caps:
Trying to connect an already connected site:
![image](https://cloud.githubusercontent.com/assets/1554855/13879074/bb61cdee-ed15-11e5-8ab0-093e6689fd40.png)

Trying to connect a site with a deactivated jetpack:
![image](https://cloud.githubusercontent.com/assets/1554855/13879097/d5713fa8-ed15-11e5-8c43-33d28ac73ccf.png)

Trying to connect a site with no jetpack:
![image](https://cloud.githubusercontent.com/assets/1554855/13879105/ea7016ae-ed15-11e5-9b78-d5f8cdf9bd5d.png)

Non-existent url:
![image](https://cloud.githubusercontent.com/assets/1554855/13879116/f520ced6-ed15-11e5-9f0b-ab6c3b05d450.png)


How to test
==========
1. Go to http://calypso.localhost:3000/jetpack/connect
2. Enter different urls: non existent ones, existent but not wordpress site, wordpress sites without jetpack, worpress site with an inactive jetpack, wordpress site with an active but unconnected jetpack
3. All the urls should trigger a notice telling what the problem is, except for the site without jetpack (Should open a dialog telling you how to install it) and the active but unconnected (should start the remote auth process... if your jetpack is not using the correct branch, it will redirect you to a non-working url)


What's missing:
==========
I need to add the redux actions & reducers tests yet. This PR was already quite large, so I'll add them in the next one

ping @roccotripaldi @oskosk @ryelle @rickybanister @alternatekev
